### DISCUSSION TITLE
Adding a "read" varaible to /oepnbook when using /openbook with joinServer EventCommand.

### DISCUSSION THREAD

**Discussion by TomLewis** [Zrips/CMI#4544] [Feature Requests]
**Creating a book news when people log in**
We have /openbook to open a book, and we also have joinServer in eventCommands, so we can basically build BookNews plugin https://www.spigotmc.org/resources/booknews-1-8-1-16-4.61163/ without needing the plugin or https://www.spigotmc.org/resources/automaticbooks-book-on-join-announces-in-game-customizable.66639/ so when a user logs in it opens a book to show them the latest news, but theres 2 issues I cant figure out.

1. how would I limit the book to only be open once when they log in so its marked as read? So if they log in and out multilpe times a day it dousn't show up every time.
2. Without making a new book each time, how would I show the new updated book if it was updated?

**Comment by Zrips** [comment]
In general you cant get this working with current setup and this would require some specific handling where we would mark each book with unique id and record that player read it and we would show it again if it got some updates in it which would generate new id. So for now you might need to stick with those plugins

**Comment by TomLewis** [comment]
> 
> 
> In general you cant get this working with current setup and this would require some specific handling where we would mark each book with unique id and record that player read it and we would show it again if it got some updates in it which would generate new id. So for now you might need to stick with those plugins

Thanks for the feedback, you have so many little addons I thought it would be worth asking first! 

would this be out of scope to be added into CMI? You have thousands of tickets in here, I have no idea how you stay organised and prioritize ideas.

I was looking at some tickets from 2018 the other day in here, and a few of them have been implemented I think and could be closed.

**Comment by Zrips** [comment]
yea, i will need to go throw tickets and close those which got implemented. In general new features are prioritized by some weird criteria like being easy to implement or being fun to code. Things like custom recipes based on item name, lore and so on is not really interesting to code and would take allot of time to do, so its not prioritized. But some 15minute features can be implemented, especially if those are more wide used and not some very specific feature.

**Comment by TomLewis** [comment]
Ha, I actually figured this out from what you add in your changelogs to what I see in tickets here, that's why I always request small changes that I know will be possible with what you have already!
